app/src/main/java/org/dhis2/usescases/general/ActivityGlobalAbstract.java

```diff
@@ -88,6 +92,14 @@ public void setScreenName(String name) {
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
+        ServerComponent serverComponent = ((App) getApplicationContext()).getServerComponent();
```

Comment:
instead of add the code to the base activity, as this is related with session, would it make sense to have some kind os sessionController or sessionService and inject it where we need it in order to have this logic isolated?

Reply:
At first I implemented it with an Inject. But there were problems in the splash and login activity as they also extend the ActivityGlobalAbstract but the server component has not been created yet.